common/app/model/Badges.scala

```diff
@@ -8,19 +8,17 @@ import java.math.BigInteger
 import scala.util.control.NonFatal
 
 
-trait BaseBadge{
-  def imageUrl(id: String): String
-  def classModifier: Option[String]
-  def isBadgeTag(tag: String):Boolean
+trait BaseBadge {
+  def maybeThisBadge(tag: String): Option[Badge]
 }
 
-case class Badge(seriesTag: String, staticImageUrl: String, classModifier: Option[String] = None) extends BaseBadge {
-  def isBadgeTag(tag: String) = seriesTag == tag
```

Comment:
why have you not keep this `isBadgeForTag()` method? 
I found having a class method return `this` kind of weird. Also it mean using `flatmap` below when `find` is imho a better way to convey what the code is doing.